plugins/orchestrator-common/config.d.ts

```diff
@@ -59,6 +59,13 @@ export interface Config {
         path?: string;
       };
     };
+    dataIndexService: {
+      /**
+       * URL of the Data Index service.
+       * Example: http://localhost:8080/graphql
+       */
+      url: string;
```

Comment:
```suggestion
      /**
       * URL of the Data Index service.
       * Default: http://localhost:8080/graphql
       */
      url?: string;
```

Since this field is optional (you are using the `getOptionalString` to read it), the property must also be optional.